src/gui/editors/PianoRoll.cpp

```diff
@@ -374,6 +374,23 @@ PianoRoll::PianoRoll() :
 	connect( &m_noteLenModel, SIGNAL( dataChanged() ),
 					this, SLOT( noteLengthChanged() ) );
 
+	// Set up key selection
+	m_keyModel.addItem(tr("No key"));
+	m_keyModel.addItem(tr("C"));
+	m_keyModel.addItem(tr("C♯/D♭"));
+	m_keyModel.addItem(tr("D"));
+	m_keyModel.addItem(tr("D♯/E♭"));
+	m_keyModel.addItem(tr("E"));
+	m_keyModel.addItem(tr("F"));
+	m_keyModel.addItem(tr("F♯/G♭"));
+	m_keyModel.addItem(tr("G"));
+	m_keyModel.addItem(tr("G♯/A♭"));
+	m_keyModel.addItem(tr("A"));
+	m_keyModel.addItem(tr("A♯/B♭"));
+	m_keyModel.addItem(tr("B"));
```

Comment:
Perhaps this could be a collection `keys` and then a `for_each` could call `addItem` on them all? I'm fine with leaving it like this though.

Reply:
Yeah I was thinking about this myself. When I wrote it I felt it was short enough and never going to change so this was an okay solution. Possible area for cleanup later for sure towards DRY.